monai/data/image_reader.py

```diff
@@ -558,9 +560,13 @@ def get_data(self, img):
         for i in ensure_tuple(img):
             header = {}
             if isinstance(i, np.ndarray):
-                # can not detect the channel dim of numpy array, use all the dims as spatial_shape
-                header["spatial_shape"] = i.shape
+                # if `channel_dim` is None, can not detect the channel dim, use all the dims as spatial_shape
+                spatial_shape = np.asarray(i.shape)
+                if isinstance(self.channel_dim, int):
+                    spatial_shape = np.delete(spatial_shape, self.channel_dim)
+                header["spatial_shape"] = spatial_shape
             img_array.append(i)
+            header["original_channel_dim"] = self.channel_dim if isinstance(self.channel_dim, int) else "no_channel"
```

Comment:
not sure how this works with the `npz_keys` option, should raise an exception?

Reply:
Hi @wyli ,

I think it's same as other image readers when reading a list of files and stack them together.
It will raise exception if having channel dim:
https://github.com/Project-MONAI/MONAI/blob/dev/monai/data/image_reader.py#L130

Thanks.